### PR TITLE
Update Calico to v2.4.1 release

### PR DESCRIPTION
**Pull request by tmjd**

### CODE REVIEW

**Comment by squat** [comment]
@tmjd looks good overall. Why do we need to configure the env var `CLUSTER_TYPE`? Is that for calico telemetry?

**Comment by tmjd** [comment]
That is loaded into the datastore and can be retrieved by the `calicoctl version` command along with the version of calico/node.  Hopefully it will help provide support some initial info on how a calico cluster is deployed. We include it in telemetry too.

**Comment by abhinavdahiya** [comment]
@tmjd @squat will probably need update to rbac policy (https://github.com/coreos/tectonic-installer/pull/1578/files#diff-c8cd1895db2e87fe14ea07a2e0794585R88)
also, it would be great if we can add readiness/liveness probes for felix (https://github.com/coreos/tectonic-installer/pull/1578/files#diff-c8cd1895db2e87fe14ea07a2e0794585R201)

**Comment by tmjd** [comment]
@abhinavdahiya Sorry I didn't find your PR for the update to 2.4.0 (I've been updating several integrations and must have forgot to search here before submitting).  Should I just close mine and you can update yours to 2.4.1? (Or I can PR against your branch.)

**Comment by abhinavdahiya** [comment]
@tmjd I think it is easier to just add rbac&probes in this PR itself. :innocent: 
closes #1578

**Comment by tmjd** [comment]
Updated RBAC and added the liveness and readiness probes.

File: modules/net/calico-network-policy/resources/manifests/kube-calico.yaml [hunk 1]
```diff
@@ -86,18 +86,24 @@ rules:
...
     resources:
       - globalconfigs
+      - globalbgpconfigs
     verbs:
       - create
       - get
       - list
       - update
       - watch
+   - apiGroups: ["projectcalico.org"]
```
Comment (galexrt) [inline modules/net/calico-network-policy/resources/manifests/kube-calico.yaml]: The ident is wrong here. One space too much. I get:
```
error: error converting YAML to JSON: yaml: line 58: did not find expected '-' indicator
```
When removing one space from line 96 through 101, I can create the manifest.

Comment (tmjd) [inline modules/net/calico-network-policy/resources/manifests/kube-calico.yaml]: Sorry about that, the lines I added below there were indented one extra too so I fixed that too.

**Comment by Quentin-M** [comment]
ok to test

**Comment by Quentin-M** [comment]
ok to test

**Comment by Quentin-M** [comment]
retest this please

**Comment by Quentin-M** [comment]
retest this please

**Comment by Quentin-M** [comment]
ok to test

**Comment by Quentin-M** [comment]
Calico test failed:

```
        --- FAIL: Test/Cluster/NetworkPolicy (250.21s)
        	cluster_test.go:388: failed to create test nginx: pods in deployment nginx-dkx5v not ready: timed out waiting for the condition
        	cluster_test.go:388: failed to create test nginx: pods in deployment nginx-qf7kb not ready: timed out waiting for the condition
        	cluster_test.go:393: failed to create an testworkload: timed out waiting for the condition
```

**Comment by abhinavdahiya** [comment]
@Quentin-M that looks like nginx-testworkload for testing network policy wasn't ready.

**Comment by abhinavdahiya** [comment]
ok to test

**Comment by Quentin-M** [comment]
ok to test

**Comment by Quentin-M** [comment]
retest this please.

**Comment by abhinavdahiya** [comment]
retest this please.

**Comment by Quentin-M** [comment]
The only failure is due to Quay generating a 500. Thank you so much @abhinavdahiya. Let's try to see if it'd work with https://github.com/coreos/tectonic-installer/pull/1723 and https://github.com/coreos/tectonic-installer/pull/1699.

**Pull request merged**